### ISSUE TITLE
Not able to mock SplFileObject

### ISSUE THREAD

**Issue by ajant** [phpspec/prophecy#262]
When trying to mock SplFileObject:
`$prophetSplFileObject = $this->prophesize('\SplFileObject');`
I get a failure, when I try to reveal mock object
`$prophetSplFileObject->getSize()->shouldBeCalled()->willReturn(1);`

Error message:

> SplFileObject::__construct(): Filename cannot be empty

I'm working with 
`"phpunit/phpunit": "~4.7"`
but, I also tried using
`phpspec/prophecy (dev-master b02221e)`
same result

**Comment by aik099** [comment]
That's because Prophecy in contrast to other mocking frameworks always extends mocked class and creates instance of it. As a side effect of this there is no way to prevent constructor code of mocked class from running.

You can try using `->beConstructedWith(...)` method on a mock before revealing to specify constructor arguments.

**Comment by pamil** [comment]
It is fixed by #238, just waiting for merge.

**Comment by ajant** [comment]
@pamil  Any idea on when it will be merged?

@aik099  I don't know prophecy internals, so I can't pretend to give any really useful suggestions on how to fix any real problems, but you can instantiate object without calling constructor:

```
$ref = new \ReflectionClass('SomeClass');
$obj = $ref->newInstanceWithoutConstructor();
```

However, I have been able to mock classes that require constructor parameters before, without having to provide any info on constructor arguments, so I don't think that's the issue here. Then again, not knowing much about implementation, I might be wrong

**Comment by pamil** [comment]
@ajant don't know, it has been there for a while. You can't mock `\SplFileObject` without calling a constructor because it's built-in class.

**Comment by ajant** [comment]
Ok, thanx, for the time being, I've fixed it using:
```
$prophetSplFileObject = $this->prophesize('\SplFileObject');
$prophetSplFileObject->willBeConstructedWith([__DIR__ . '/ActualTest.php', 'r', true, null]);
```
I'll keep an eye on your pull request

**Comment by everzet** [comment]
Merged #238 into master
